### PR TITLE
[Feat](Nereids) support show variables and show view command

### PR DESCRIPTION
**Pull request by starocean999**
### What problem does this PR solve?

Issue Number: close #xxx

Related PR: #xxx

Problem Summary:
migrate ShowVariablesStmt and ShowViewStmt to nereids

### Check List (For Committer)

- Test 

    - [X] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No colde files have been changed.
        - [ ] Other reason 

- Behavior changed:

    - [X] No.
    - [ ] Yes. 

- Does this need documentation?

    - [X] No.
    - [ ] Yes. 

- Release note

    
    
    None

### Check List (For Reviewer who merge this PR)

- [X] Confirm the release note
- [X] Confirm test cases
- [X] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by starocean999** [comment]
run buildall

**Comment by starocean999** [comment]
run buildall

**Comment by starocean999** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowCommand.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import org.apache.doris.analysis.StmtType;
+import org.apache.doris.nereids.trees.plans.PlanType;
+import org.apache.doris.qe.ConnectContext;
+import org.apache.doris.qe.ShowResultSet;
+import org.apache.doris.qe.StmtExecutor;
+
+/**
+ * base class for all show commands
+ */
+public abstract class ShowCommand extends Command implements NoForward {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowCommand.java:29]: why ShowCommand implement NoForward?

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowVariablesCommand.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    private static final String VALUE_COL = "Value";
+    private static final String DEFAULT_VALUE_COL = "Default_Value";
+    private static final String CHANGED_COL = "Changed";
+    private static final ShowResultSetMetaData META_DATA = ShowResultSetMetaData.builder()
+            .addColumn(new Column(NAME_COL, ScalarType.createVarchar(20)))
+            .addColumn(new Column(VALUE_COL, ScalarType.createVarchar(20)))
+            .addColumn(new Column(DEFAULT_VALUE_COL, ScalarType.createVarchar(20)))
+            .addColumn(new Column(CHANGED_COL, ScalarType.createVarchar(20)))
+            .build();
+    private SetType type;
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowVariablesCommand.java:50]:
```suggestion
            .build();
            
    private final SetType type;
```

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowVariablesCommand.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    private String pattern;
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowVariablesCommand.java:51]:
```suggestion
    private final String pattern;
```

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowViewCommand.java [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+        this.tbl = tbl;
+    }
+
+    @Override
+    public <R, C> R accept(PlanVisitor<R, C> visitor, C context) {
+        return visitor.visitShowViewCommand(this, context);
+    }
+
+    private void validate(ConnectContext ctx) throws Exception {
+        if (tbl == null) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowViewCommand.java:75]: why tbl could be null? i think in syntax, we always have a tbl. so we should check not null when ctor?

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowViewCommand.java [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+            ErrorReport.reportAnalysisException(ErrorCode.ERR_NO_TABLES_USED);
+        }
+        if (!Strings.isNullOrEmpty(db)) {
+            // if user specify the `from db`, overwrite the db in `tbl` with this db.
+            // for example:
+            //      show view from db1.tbl1 from db2;
+            // will be rewrote to:
+            //      show view from db2.tbl1;
+            // this act same as in MySQL
+            tbl.setDb(db);
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowViewCommand.java:85]: in syntax, we use multi identifier as table name. if we should set db on tbl, should we only allow identifier as table name? what does `from db` really mean? it indicate table from this db or view from this db? how to show db_a.views that use db_b.tbl?

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowCommand.java (hunk 1 above)
Comment (starocean999) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowCommand.java:29]: all subclass of ShowStmt are NO_FORWARD in old planner, so we keep same logic here

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowViewCommand.java (hunk 4 above)
Comment (starocean999) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowViewCommand.java:75]: removed

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowViewCommand.java (hunk 5 above)
Comment (starocean999) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowViewCommand.java:85]: keep the behavior same as old planner

**Comment by starocean999** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Review by englefly — APPROVED** [review]

**Pull request merged**